packages/payment-processor/src/payment/single-request-proxy.ts

```diff
@@ -0,0 +1,187 @@
+ * It uses the SingleRequestProxyFactory contract to create either an ERC20 or Ethereum Single Request Proxy.
+ * The function calculates the payment reference and handles the deployment transaction, including waiting for confirmation.
+ * The factory address is automatically determined based on the payment chain using the singleRequestProxyFactoryArtifact.
+ */
+export async function deploySingleRequestProxy(
+  request: ClientTypes.IRequestData,
+  signer: Signer,
+): Promise<string> {
+  const requestPaymentNetwork = getPaymentNetworkExtension(request);
+
+  // Check if the payment network is supported, only ERC20_FEE_PROXY_CONTRACT and ETH_FEE_PROXY_CONTRACT are supported
+  if (
+    !requestPaymentNetwork ||
+    (requestPaymentNetwork.id !== ExtensionTypes.PAYMENT_NETWORK_ID.ERC20_FEE_PROXY_CONTRACT &&
+      requestPaymentNetwork.id !== ExtensionTypes.PAYMENT_NETWORK_ID.ETH_FEE_PROXY_CONTRACT)
+  ) {
+    throw new Error('Unsupported payment network');
+  }
+
+  const paymentChain = request.currencyInfo.network;
+  if (!paymentChain) {
+    throw new Error('Payment chain not found');
+  }
+
+  // Use artifact's default address for the payment chain
+  const singleRequestProxyFactory = singleRequestProxyFactoryArtifact.connect(
+    paymentChain as CurrencyTypes.EvmChainName,
+    signer,
+  );
+
+  if (!singleRequestProxyFactory.address) {
+    throw new Error(`SingleRequestProxyFactory not found on chain ${paymentChain}`);
+  }
+
+  const salt = requestPaymentNetwork?.values?.salt;
+  const feeAddress = requestPaymentNetwork?.values?.feeAddress;
+  const feeAmount = requestPaymentNetwork?.values?.feeAmount;
+  const paymentRecipient = requestPaymentNetwork?.values?.paymentAddress;
+
+  if (!salt || !feeAddress || !feeAmount || !paymentRecipient) {
+    throw new Error('Invalid payment network values');
+  }
+
+  const paymentReference = `0x${PaymentReferenceCalculator.calculate(
+    request.requestId,
+    salt,
+    paymentRecipient,
+  )}`;
+
+  const isERC20 =
+    requestPaymentNetwork.id === ExtensionTypes.PAYMENT_NETWORK_ID.ERC20_FEE_PROXY_CONTRACT;
+
+  let tx;
+
+  if (isERC20) {
+    const tokenAddress = request.currencyInfo.value;
+    tx = await singleRequestProxyFactory.createERC20SingleRequestProxy(
+      paymentRecipient,
+      tokenAddress,
+      paymentReference,
+      feeAddress,
+      feeAmount,
+    );
+  } else {
+    tx = await singleRequestProxyFactory.createEthereumSingleRequestProxy(
+      paymentRecipient,
+      paymentReference,
+      feeAddress,
+      feeAmount,
+    );
+  }
+
+  const receipt = await tx.wait();
+
+  const event = receipt.events?.[0];
+
+  if (!event) {
+    throw new Error('Single request proxy creation event not found');
+  }
+
+  const proxyAddress = ethers.utils.defaultAbiCoder.decode(['address', 'address'], event.data)[0];
+
+  if (!proxyAddress) {
+    throw new Error('Proxy address not found in event data');
+  }
```

Comment:
@aimensahnoun why did you change the event handling logic? 

The code that CodeRabbit commented is very similar to what you had before.

Reply:
Nevermind. I understand now that since the argument isn't indexed, it's encoded in the data and is no longer present in the args. Thus, CodeRabbit's comment is incorrect.

Reply:
Resolving as No Problem :ok_person: